tests/agent/test_agent.py

```diff
@@ -294,31 +319,34 @@ def test_correlation_id_header(
         rsps.add(
             responses.GET,
             f"{base_url}/organizations/{org_id}/data-context-configuration",
-            json={
-                "anonymous_usage_statistics": {
-                    "data_context_id": str(uuid.uuid4()),
-                    "enabled": False,
-                },
-                "datasources": {},
-                "stores": {},
-            },
+            json=data_context_config,
         )
         rsps.add(
             responses.GET,
             f"{base_url}/organizations/{org_id}/datasources/drafts/{datasource_config_id_1}",
-            json={},
+            json={"data": {"attributes": {"draft_config": ds_config_factory("test-ds-1")}}},
             # match will fail if correlation-id header is not set
-            match=[
-                responses.matchers.header_matcher({HeaderName.AGENT_JOB_ID: str(agent_job_id_1)})
-            ],
+            match=[responses.matchers.header_matcher({HeaderName.AGENT_JOB_ID: agent_job_ids[0]})],
         )
         rsps.add(
             responses.GET,
             f"{base_url}/organizations/{org_id}/datasources/drafts/{datasource_config_id_2}",
-            json={},
+            json={"data": {"attributes": {"draft_config": ds_config_factory("test-ds-2")}}},
+            # match will fail if correlation-id header is not set
+            match=[responses.matchers.header_matcher({HeaderName.AGENT_JOB_ID: agent_job_ids[1]})],
+        )
+        rsps.add(
+            responses.GET,
+            f"{base_url}organizations/{org_id}/checkpoints/{checkpoint_id}",
+            json={"data": {}},
             # match will fail if correlation-id header is not set
             match=[
-                responses.matchers.header_matcher({HeaderName.AGENT_JOB_ID: str(agent_job_id_2)})
+                responses.matchers.header_matcher(
+                    {
+                        HeaderName.USER_AGENT: f"{USER_AGENT_HEADER}/{GXAgent._get_current_gx_agent_version()}",  # TODO: remove
+                        HeaderName.AGENT_JOB_ID: agent_job_ids[2],
+                    }
```

Comment:
Note: `User-Agent` is being set but `Agent-Job-Id` is not.
But this only happens for `RunCheckpointEvent`; for `DraftDatasourceConfigEvent`, it's set without issue.